packages/common/src/utils/signature_utils.ts

```diff
@@ -87,4 +88,27 @@ export class SignatureUtils {
 
     return cacao
   }
+
+  /**
+   * Takes a StreamState and validates that none of the commits in its log are based on expired CACAOs.
+   */
+  static checkForCacaoExpiration(state: StreamState): void {
+    const now = Math.floor(Date.now() / 1000) // convert millis to seconds
+    for (const logEntry of state.log) {
+      const timestamp = logEntry.timestamp ?? now
+      if (!logEntry.expirationTime) {
+        continue
+      }
+      const expirationTime = logEntry.expirationTime + DEFAULT_CACAO_REVOCATION_PHASE_OUT_SECS
```

Comment:
mostly this function is just moved from StreamUtils into this file, the only functional change is adding the `DEFAULT_CACAO_REVOCATION_PHASE_OUT_SECS` here.